### ISSUE TITLE
Add support for .pvr.ccz (including pvrtc formats)

### ISSUE THREAD

**Issue by cjhanson** [cocos2d/CocosBuilder#111]
Please sign up for the PowerVR Insider here: http://www.imgtec.com/powervr/insider/sdkdownloads/index.asp

Then you can get the sdk which has some libraries you can include with cocosbuilder to allow for reading PVR files. Their license is permissive and just requires a small acknowledgement in the documentation of cocosbuilder. To support pvrtc on mac it will likely mean reading in the image and handing converting it at load time into an uncompressed texture format that mac can render.

**Comment by ebrake** [comment]
Second this! I have to export my textures a second time (adding considerable workflow) in a non-PVRTC format just to see them in CocosBuilder, which shows PVRTC images as solid black boxes.

**Comment by mikey0000** [comment]
I have pvr.ccz files working in coocsbuilder

make sure you specify in the plist under metadata
realTextureFileName : yourpvrfile
textureFileName :yourpvrfile

**Comment by mikey0000** [comment]
From downloading the code and having a look it uses cocos2d so theoretically it should support anything cocos2d does. BTW this is from 2.1 rc3 that I tested

**Comment by cjhanson** [comment]
@mikey0000 any chance you have texture packer installed? I think it installs some helper stuff at the OS level for handling pvr and pvr.ccz images.

Can anyone confirm that .pvr and .pvr.ccz images DO NOT render if you don't have Texture Packer?

**Comment by mikey0000** [comment]
Umm I don't think so, I have both texture packer and sprite helper, I only used texture packer to make the pvrs however I didn't make them in pvrtc format they are rgba4444. They didn't work if I didn't have what I mentioned above in the plists even if I used texture packer. I'm happy to do some testing of any sort. I highly doubt that texture packer installs anything that cocosbuilder would use in its code base that would help due to the scene part of cocosbuilder using cocos2d to draw everything and since cocos2d has support for pvr formats it works.

**Comment by ebrake** [comment]
I can confirm that PVRTC4 (bundled as .pvr.ccz) do not display for me in the current CocosBuilder regardless of plist settings.

**Comment by mikey0000** [comment]
I'll try with pvrtc format and see what happens, I'm guessing your plist is named the same too? Can you try with pvr with rgba4444?

**Comment by mikey0000** [comment]
ok also on my side using pvrtc formats do not work, I have modified cocosbuilder to accept pvr files and can test them directly without plist files as to whether or not they work. From what I have found although you can't see them it is supported on the iPhone from cocosbuilder. From what I have read from the imgtec website to get PVRTC textures to display on a mac requires decompression (which is possible) before being displayed as PVRTC is not supported natively on a mac due to not having powervr graphics gear.

also see here http://www.imgtec.com/forum/forum_posts.asp?TID=1130&title=pvrtc-on-mac

the post is a bit old but you'll get the idea.

summed up
any PVRTC format is not supported on the mac.
however RGBA\* formats in PVR compression are.

please let me know if I am wrong at all.

**Comment by ebrake** [comment]
You are correct, PowerVR decompression support would need to be added to the project for PVRTC textures to display in CocosBuilder!

**Comment by cjhanson** [comment]
That's what I was saying in the first place. It will need to make use of the powervr sdk.

**Comment by ricardoquesada** [comment]
+1 for this feature

**Comment by dominik-hadl** [comment]
+1 also. This would be very nice to have.

**Comment by mrsergio** [comment]
+1 too. Need this feature.

**Comment by young40** [comment]
+1, need this feature.
